controllers/release/adapter.go

```diff
@@ -480,6 +619,146 @@ func (a *adapter) cleanupProcessingResources(pipelineRun *tektonv1.PipelineRun,
+			previousRelease.Namespace, types.Separator, previousRelease.Name)
+	}
+
+	return utils.NewPipelineRunBuilder(pipelineType, namespace).
+		WithAnnotations(metadata.GetAnnotationsWithPrefix(a.release, integrationgitops.PipelinesAsCodePrefix)).
+		WithFinalizer(metadata.ReleaseFinalizer).
+		WithLabels(map[string]string{
+			metadata.PipelinesTypeLabel:    pipelineType,
+			metadata.ReleaseNameLabel:      a.release.Name,
+			metadata.ReleaseNamespaceLabel: a.release.Namespace,
+		}).
+		WithObjectReferences(a.release).
+		WithOwner(a.release).
+		WithParams(
+			tektonv1.Param{
+				Name: "previousRelease",
+				Value: tektonv1.ParamValue{
+					Type:      tektonv1.ParamTypeString,
+					StringVal: previousReleaseNamespaceName,
+				},
+			},
+		).
+		WithPipelineRef((&utils.PipelineRef{
+			Resolver: "git",
+			Params: []utils.Param{
+				{
+					Name:  "url",
+					Value: "https://github.com/konflux-ci/release-service-catalog.git",
+				},
+				{
+					Name:  "revision",
+					Value: revision,
+				},
+				{
+					Name:  "pathInRepo",
+					Value: "pipelines/run-collectors/run-collectors.yaml",
+				},
+			},
+		}).ToTektonPipelineRef()).
+		WithWorkspaceFromVolumeTemplate(
+			os.Getenv("DEFAULT_RELEASE_WORKSPACE_NAME"),
+			os.Getenv("DEFAULT_RELEASE_WORKSPACE_SIZE"),
+		)
+}
+
+// createManagedCollectorsPipelineRun creates a PipelineRun to run the collectors Pipeline for collectors in the ReleasePlanAdmission.
+func (a *adapter) createManagedCollectorsPipelineRun(releasePlanAdmission *v1alpha1.ReleasePlanAdmission) (*tektonv1.PipelineRun, error) {
+	revision, err := releasePlanAdmission.Spec.Pipeline.PipelineRef.GetRevision()
+	if err != nil {
+		revision = "production"
+	}
+
+	var pipelineRun *tektonv1.PipelineRun
+	pipelineRun, err = a.getCollectorsPipelineRunBuilder(metadata.ManagedCollectorsPipelineType, releasePlanAdmission.Namespace, revision).
+		WithParams(
+			tektonv1.Param{
+				Name: "collectorsResourceType",
+				Value: tektonv1.ParamValue{
+					Type:      tektonv1.ParamTypeString,
+					StringVal: "releaseplanadmission",
+				},
+			},
+			tektonv1.Param{
+				Name: "collectorsResource",
+				Value: tektonv1.ParamValue{
+					Type: tektonv1.ParamTypeString,
+					StringVal: fmt.Sprintf("%s%c%s",
+						releasePlanAdmission.Namespace, types.Separator, releasePlanAdmission.Name),
+				},
+			},
+		).
+		WithServiceAccount(releasePlanAdmission.Spec.Collectors.ServiceAccountName).
+		Build()
+
+	if err != nil {
+		return nil, err
+	}
+
+	err = a.client.Create(a.ctx, pipelineRun)
+	if err != nil {
+		return nil, err
+	}
+
+	return pipelineRun, nil
+}
+
+// createTenantCollectorsPipelineRun creates a PipelineRun to run the collectors Pipeline for collectors in the ReleasePlan.
+func (a *adapter) createTenantCollectorsPipelineRun(releasePlan *v1alpha1.ReleasePlan, releasePlanAdmission *v1alpha1.ReleasePlanAdmission) (*tektonv1.PipelineRun, error) {
+	var revision string
+	var err error
+
+	if releasePlanAdmission != nil {
```

Comment:
if you do not have a RPA, then revision is empty and the empty param value is passed....which is incorrect.

Reply:
True. Addressed.